UltraDNS has this crazy DNS packet which Deadwood refused to resolve:

$ askmara Awww.getcafecup.com. 204.74.108.1
# Querying the server with the IP 204.74.108.1
# Question: Awww.getcafecup.com.
WWW.getcafecup.com. +7200 a 216.242.93.163
# NS replies:
#getcafecup.com. +3600 ns pdns1.ultradns.net.
#getcafecup.com. +3600 ns pdns2.ultradns.net.
#getcafecup.com. +3600 ns pdns5.ultradns.info.
#getcafecup.com. +3600 ns pdns6.ultradns.co.uk.
#getcafecup.com. +3600 ns pdns4.ultradns.org.
#getcafecup.com. +3600 ns pdns3.ultradns.org.
# AR replies:

This patch makes these resolve in Deadwood.

Yes, the RFC allows this *but* it's better to perserve the case
of the incoming query because doing so makes the compressed DNS packet 
smaller.

--- deadwood-3.2.04/src/DwRecurse.c.orig	2013-07-13 15:23:10.955049409 -0700
+++ deadwood-3.2.04/src/DwRecurse.c	2013-07-13 15:24:29.806043909 -0700
@@ -170,6 +170,7 @@
 int dwx_dname_issame(uint8_t *p1, uint8_t *p2, uint8_t *pmax1,
                 uint8_t *pmax2) {
         int len = 0, counter = 0;
+	uint8_t lc1, lc2; /* Lower case */
 
         if(*p1 != *p2) {
                 return 0;
@@ -191,7 +192,11 @@
                         if(p1 > pmax1 || p2 > pmax2) {
                                 return -1;
                         }
-                        if(*p1 != *p2) {
+			lc1 = *p1;
+			lc2 = *p2;
+			if(lc1 >= 'A' && lc1 <= 'Z') { lc1 += 32; }
+			if(lc2 >= 'A' && lc2 <= 'Z') { lc2 += 32; }
+                        if(lc1 != lc2) {
                                 return 0;
                         }
                 }
